IdentityCore/src/workplacejoin/MSIDWorkPlaceJoinUtilBase.m

```diff
@@ -147,12 +151,20 @@ For asymmetric keys this holds the public key hash which allows digital identity
     }
     
     NSMutableDictionary *mutableCertQuery = [NSMutableDictionary new];
-    
     if (certAttributes)
     {
         [mutableCertQuery addEntriesFromDictionary:certAttributes];
     }
     
+#if TARGET_OS_OSX
+    // For macOS, if the key is ECC key, use shared access group to query certificate. If it is RSA, remove shared access group from query for certificate as it is not login keychain.
+        NSString *sharedAccessGroup = [certAttributes valueForKey:(__bridge id)kSecAttrAccessGroup];
+        if (sharedAccessGroup && ![[MSIDKeyOperationUtil sharedInstance] isKeyFromSecureEnclave:privateKeyRef])
+        {
+            [mutableCertQuery removeObjectForKey:(__bridge id)kSecAttrAccessGroup];
```

Comment:
What about private key query? Shouldn't we also remove shared group attribute there?

Reply:
Yes that is correct, thanks. Updated code to remove access group for macOS legacy registration check